backend/algorithms/objects/conditions.py

```diff
@@ -156,41 +166,74 @@ def get_warning(self, applicable_wam):
             return None
         return f"Requires {self.wam} WAM in {self.category}. Your WAM in {self.category} is currently {applicable_wam:.3f}"
 
+    def __str__(self) -> str:
+        return f"{self.wam}WAM in {self.category}"
+
 
-class GRADECondition(Condition):
-    """Handles GRADE conditions such as 65GRADE and 80GRADE in [A-Z]{4}[0-9]{4}"""
+class GradeCondition(Condition):
+    """ Handles Grade conditions such as 65GRADE and 80GRADE in [A-Z]{4}[0-9]{4} """
 
-    def __init__(self, grade, course):
+    def __init__(self, grade: int):
         self.grade = grade
+        self.category = CompositeCategory()
 
-        # Course code
-        self.course = course
+    def set_category(self, category_classobj: Category):
+        """ Set own category to the one given """
+        self.category = category_classobj
+
+    def validate(self, user: User) -> tuple[bool, list[str]]:
+        def _validate_course(course: Category):
+            # Grade condition can only be used with ClassCategory
+            if type(course) is CompositeCategory:
```

Comment:
is this not guaranteed?

Reply:
no because you can have a CompositeCategory of CompositeCategories

Reply:
yup makes sense